cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -195,11 +193,11 @@ public void dropAllInNamespace(final NamespaceId namespace) throws Exception {
     });
 
     for (final Location location : locations) {
-      doDrop(namespace.stream(StreamUtils.getStreamNameFromLocation(location)), location, ugi);
+      doDrop(namespace.stream(StreamUtils.getStreamNameFromLocation(location)), location);
     }
 
     // Also drop the state table
-    ImpersonationUtils.doAs(ugi, new Callable<Void>() {
+    impersonator.doAs(namespace, new Callable<Void>() {
```

Comment:
I don't think we should be impersonating here for the namespace. In the Callable we do
```java
stateStoreFactory.dropAllInNamespace(namespace);
```
Which does
```java
  @Override
  public synchronized void dropAllInNamespace(NamespaceId namespace) throws IOException {
    try (HBaseDDLExecutor executor = ddlExecutorFactory.get(); HBaseAdmin admin = new HBaseAdmin(hConf)) {
      TableId tableId = StreamUtils.getStateStoreTableId(namespace);
      TableId hbaseTableId = tableUtil.createHTableId(new NamespaceId(tableId.getNamespace()), tableId.getTableName());
      if (tableUtil.tableExists(admin, hbaseTableId)) {
        tableUtil.dropTable(executor, hbaseTableId);
      }
    }
  }
```
Since with App level impersonation we can have Streams which is owned by a specific owner we will have a scenario in which different streams are owned by different owners in the same namespace. Since we are deleting the associated Hbase table which belong to stream table shouldn't we impersonating stream owner of the stream. Will the delete work if we impersonate the namespace owner ?
Also, is it acceptable from the perspective of the security that a user who has ADMIN on namespace can impersonate other owners of streams and delete their streams ?
I think once we have authorization to determine who can impersonate whom this question will get answered as a consequence.